feature-wallet-impl/src/main/java/io/novafoundation/nova/feature_wallet_impl/data/repository/RealTransactionHistoryRepository.kt

```diff
@@ -60,21 +67,31 @@ class RealTransactionHistoryRepository(
             filters = filters,
             accountId = accountId,
             chain = chain,
-            chainAsset = chainAsset
+            chainAsset = chainAsset,
+            currency = currency
         )
     }
 
-    override fun operationsFirstPageFlow(
+    override suspend fun operationsFirstPageFlow(
         accountId: AccountId,
         chain: Chain,
-        chainAsset: Chain.Asset
+        chainAsset: Chain.Asset,
+        currency: Currency
     ): Flow<DataPage<Operation>> {
         val accountAddress = chain.addressOf(accountId)
         val historySource = historySourceFor(chainAsset)
 
         return operationDao.observe(accountAddress, chain.id, chainAsset.id)
-            .mapList {
-                mapOperationLocalToOperation(it, chainAsset)
+            .transform { operations ->
+                emit(mapOperations(operations, chainAsset, emptyList()))
```

Comment:
Lets use named arguments when its not clear what passed value represents (`emptyList()`)